scalac-scoverage-runtime/js/src/main/scala/scalajssupport/File.scala

```diff
@@ -0,0 +1,100 @@
+package scalajssupport
+
+import scala.scalajs.js
+
+class File(path: String) {
+  def this(path: String, child: String) = {
+    this(File.nodePath.join(path, child))
+  }
+
+  def delete(): Unit = {
+    if (isDirectory()) File.fs.rmdirSync(path)
+    else File.fs.unlinkSync(path)
+  }
+
+  def getAbsolutePath(): String = {
+    File.fs.realpathSync(path)
+  }
+
+  def getName(): String = {
+    File.nodePath.basename(path)
+  }
+
+  def getPath(): String = {
+    path
+  }
+
+  def isDirectory(): Boolean = {
+    try {
+      File.fs.lstatSync(path).isDirectory()
+    }
+    catch {
+      // TODO find out what's the best way to deal with exceptions here
+      // (likely not like this)
+      case e: Exception => false
+    }
+  }
+
+  def mkdirs(): Unit = {
+    path.split("/").foldLeft("")((acc: String, x: String) => {
+      val new_acc = File.nodePath.join(acc, x)
+      try {
+        File.fs.mkdirSync(new_acc)
+      } catch {
+        case e: Exception =>
+      }
+      new_acc
+    })
+  }
+
+  def listFiles(): Array[File] = {
+    // Doesn't seem really efficient, need to check if there's a better way to transform js.Array
+    val files = File.fs.readdirSync(path)
+    val filesArray = new Array[File](files.length)
+    for ((item, i) <- filesArray.zipWithIndex) {
+      filesArray(i) = new File(File.nodePath.join(this.getPath(), files(i)))
+    }
+    filesArray
+  }
+
+  def listFiles(filter: FileFilter): Array[File] = {
+    listFiles().filter(filter.accept)
+  }
+
+  def readFile(): String = {
+    File.fs.readFileSync(path, js.Dynamic.literal(encoding="utf-8"))
+  }
+
+}
+
+trait FileFilter {
+  def accept(file: File): Boolean
+}
+
+trait FSStats extends js.Object {
+  def isDirectory(): Boolean = js.native
+}
+
+trait FS extends js.Object {
+  def closeSync(fd: Int): Unit = js.native
+  def lstatSync(path: String): FSStats = js.native
+  def mkdirSync(path: String): Unit = js.native
+  def openSync(path: String, flags: String): Int = js.native
+  def realpathSync(path: String): String = js.native
+  def readdirSync(path: String): js.Array[String] = js.native
+  def readFileSync(path: String, options:js.Dynamic): String = js.native
+  def rmdirSync(path: String): Unit = js.native
+  def unlinkSync(path: String): Unit = js.native
+  def writeSync(fd: Int, csq: CharSequence): Unit = js.native
```

Comment:
`csq: String`

`CharSequence` doesn't make sense to JavaScript. Only `String` does.